### PR TITLE
L10nreg2 async

### PR DESCRIPTION
**Pull request by djg**
This crate also adds a `Pin`-safe helper data structure `ChunkyVec`. We should decide how this should be distributed. Either with fluent or separately.

### CODE REVIEW

File: fluent-fallback/src/lib.rs [hunk 1]
```diff
@@ -1,72 +1,257 @@
...
+            generator,
+        }
+    }
+
+    pub fn on_change(&mut self) {
+        // This invalidates the cache by recreating it.
+        self.bundles = Cache::new(self.generator.bundles_sync(self.resource_ids.clone()));
+    }
+
+    fn _format_value_sync_opt<'l>(
```
Comment (zbraniecki) [inline fluent-fallback/src/lib.rs]: I don't think you need to name it with `_` - rust knows its private because it doesn't have `pub`.

**Review by zbraniecki — APPROVED** [review]

File: fluent-fallback/src/lib.rs (hunk 1 above)
Comment (djg) [reply fluent-fallback/src/lib.rs]: Fixed.

**Comment by Manishearth** [comment]
Is there a reason elsa is insufficient? Or just performance?

**Comment by zbraniecki** [comment]
I rebased that work with @djg's work on top of master here: https://github.com/projectfluent/fluent-rs/pull/201

**Pull request closed without merging**